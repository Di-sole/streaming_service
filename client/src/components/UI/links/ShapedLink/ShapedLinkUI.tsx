import { ReactNode } from 'react';
import Link from 'next/link';
import styles from './ShapedLinkUI.module.scss';

interface LinkUIProps {
    children: ReactNode;
    href: string;
    shape: 'rectangular' | 'square' | 'round';
    className?: string;
    target?: string;
}

const ShapedLinkUI = ({ children, href, shape, className, target }: LinkUIProps) => {
    const linkFormClass =
        shape === 'round'
            ? styles.container_round
            : shape === 'square'
            ? styles.container_square
            : styles.container_rectangular;

    return (
        <Link href={href} className={[styles.container, linkFormClass, className].join(' ')} target={target}>
            <div className={styles.container__content}>{children}</div>
        </Link>
    );
};

export default ShapedLinkUI;
