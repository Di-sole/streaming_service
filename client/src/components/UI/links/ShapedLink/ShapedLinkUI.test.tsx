import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Image from 'next/image';
import ShapedLinkUI from './ShapedLinkUI';
import './ShapedLinkUI.module.scss';
import footerStyles from '../../../footer/Footer.module.scss';
import devicesIcon from '../../../../public/icons/link_icons/devices_all.svg';

describe('ShapedLinkUI SNAPSHOT TESTS', () => {
    test('Round link should not be changed', () => {
        render(
            <ShapedLinkUI href="https://t.me/official_iviru" shape="round">
                <Image
                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
                    height={16}
                    width={16}
                    alt="Логотип Telegram"
                />
            </ShapedLinkUI>,
        );
        const link = screen.getByRole('link');

        expect(link).toMatchSnapshot();
    });

    test('Rectangular link should not be changed', () => {
        render(
            <ShapedLinkUI href="#!" shape="rectangular">
                <div className={footerStyles.container__textSingle}>
                    <Image src={devicesIcon} height={20} width={20} alt="Иконка умных устройств" />
                    Все устройства
                </div>
            </ShapedLinkUI>,
        );
        const link = screen.getByRole('link');

        expect(link).toMatchSnapshot();
    });
});
