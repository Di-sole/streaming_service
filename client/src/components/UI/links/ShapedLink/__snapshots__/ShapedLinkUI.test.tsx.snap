// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShapedLinkUI SNAPSHOT TESTS Rectangular link should not be changed 1`] = `
<a
  class="container container_rectangular "
  href="#!"
>
  <div
    class="container__content"
  >
    <div
      class="container__textSingle"
    >
      <img
        alt="Иконка умных устройств"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      Все устройства
    </div>
  </div>
</a>
`;

exports[`ShapedLinkUI SNAPSHOT TESTS Round link should not be changed 1`] = `
<a
  class="container container_round "
  href="https://t.me/official_iviru"
>
  <div
    class="container__content"
  >
    <img
      alt="Логотип Telegram"
      data-nimg="1"
      decoding="async"
      height="16"
      loading="lazy"
      src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
      style="color: transparent;"
      width="16"
    />
  </div>
</a>
`;
